juju/client/connection.py

```diff
@@ -287,7 +287,7 @@ async def connect(
         # _connect_with_redirect method, but create them here
         # as a reminder that they will exist.
         self.addr = None
-        self.ws = None
+        self._ws = None
```

Comment:
This would be classed as a breaking change i.e. we're making something public, private. For now, let's move this back to public.